modules/auxiliary/gather/ldap_esc_vulnerable_cert_finder.rb

```diff
@@ -78,21 +77,19 @@ def parse_dacl_or_sacl(acl)
 
       # To decode the ObjectType we need to do another query to CN=Configuration,DC=daforest,DC=com
       # and look at either schemaIDGUID or rightsGUID fields to see if they match this value.
-      next unless ace_body[:flags] && ace_body[:flags][:ace_object_type_present] == 1
+      if (object_type = ace_body[:object_type]) && !(object_type == CERTIFICATE_ENROLLMENT_EXTENDED_RIGHT || object_type == CERTIFICATE_AUTOENROLLMENT_EXTENDED_RIGHT)
```

Comment:
Discovered a slight bug here upon further testing. If `ace_body[:object_type]` is `nil` then the first half of this equation will fail. However the current statement requires both sides to fail for it to work. So when we go to evaluate the second half of the equation we realize that `object_type == CERTIFICATE_ENROLLMENT_EXTENDED_RIGHT` will also return `false`, however this will get converted to `true` by the `!()` part of the code.

So therefore this statement will evaluate to `false && true` which will prevent the `next` statement inside this block from executing.

I think we likely need to redesign this logic a bit. The code on the first half should fail if the object_type field is `nil`, whilst the second half should fail if the object type is not what is expected. Given this, it may be better to rewrite this as the following line:

```
unless (object_type = ace_body[:object_type]) && (object_type == CERTIFICATE_ENROLLMENT_EXTENDED_RIGHT || object_type == CERTIFICATE_AUTOENROLLMENT_EXTENDED_RIGHT)
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/17968/commits/0441c1f18760ddc876736ddd138d1bee9a91f35c

Reply:
>  If ace_body[:object_type] is nil then the first half of this equation will fail.

That is the core of the intended change. We need to process ACEs that are not associated with object types. Before these changes we processed only ACEs associated with those two objects. Now we need to process ACEs that are associated with those two objects, **or no object at all**. 

Both before and after we should be skipping ACEs that are associated with an object other than those two.

This change should be reverted.

Reply:
Change was reverted after further discussions over Zoom. Commit was originally added due to a misunderstanding of the intent of the patch which was further clarified via said discussion and the comment above.